src/theory/quantifiers/sygus/term_database_sygus.cpp

```diff
@@ -371,7 +371,9 @@ void TermDbSygus::registerSygusType( TypeNode tn ) {
             Trace("sygus-db") << ", kind = " << sk;
             d_kinds[tn][sk] = i;
             d_arg_kind[tn][i] = sk;
-          }else if( sop.isConst() ){
+          }
+          else if (sop.isConst() && dt[i].getNumArgs() == 0)
```

Comment:
I'm a bit puzzled by this. When `sop.isConst()` is true shouldn't we consider it a constant even if the constructor has arguments?

Reply:
One example where this was wrong is BV extract operators, whose operators are constant, not builtin, but applications of these are not constants.